Compiling usd-qt with USD 19.11 and use with maya-usd

### Issue

I'm trying to compile `usd-qt` with the `dev` branch to test it together with [`maya-usd`](https://github.com/Autodesk/maya-usd). 

The latest [`maya-usd/refactoring_sandbox`](https://github.com/Autodesk/maya-usd/tree/refactoring_sandbox) and [`maya-usd/dev`](https://github.com/Autodesk/maya-usd/tree/dev) branches now build on `USD` 19.11. Basically [this USD commit](https://github.com/PixarAnimationStudios/USD/commit/1d08054341653e46cbb7120f65cc97c43133775d) in the `USD/dev`  branch as described [here](https://github.com/Autodesk/maya-usd/blob/e47ca0229dc60158c0a87acbbb404df9117be350/doc/DEVELOPER.md#source-versions). 

However I'm unable to compile `usd-qt` with that USD version.
I ran `cmake` using a somewhat equivalent of:
```bat
:: USD_ROOT is the compiled USD root.
set USD_ROOT=D:\dev\usd\build

:: Dependencies
set TBB_INCLUDE_DIR=%USD_ROOT%/src/tbb2017_20170226oss
set Boost_INCLUDE_DIR=%USD_ROOT%/include/boost-1_65_1

:: Output directory
set BUILD_DIR=D:\dev\usd\usd-qt-build

:: Cmake build
cd /D D:\dev\usd\usd-qt
cmake -G "Visual Studio 15 2017 Win64" -DBoost_INCLUDE_DIR="%Boost_INCLUDE_DIR%" -DTBB_ROOT_DIR="%TBB_INCLUDE_DIR%" -DBOOST_LIBRARYDIR=%USD_ROOT%/lib -B %BUILD_DIR%
```

And then build the `usdqt.sln` solution using Visual Studio.

Which gave me this output:
[usdqt_build_release_log.txt](https://github.com/LumaPictures/usd-qt/files/3794759/usdqt_build_release_log.txt)

Note the build failure and the errors that occurred, like:
```
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(90): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(90): error C2143: syntax error: missing ',' before '&'
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(92): error C4430: missing type specifier - int assumed. Note: C++ does not support default-int
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(92): error C2143: syntax error: missing ',' before '&'
```

And:
```
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(90): error C3668: 'pxrInternal_v0_19__pxrReserved__::UsdQt_UndoStateDelegate::_OnSetField': method with override specifier 'override' did not override any base class methods
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(98): error C3668: 'pxrInternal_v0_19__pxrReserved__::UsdQt_UndoStateDelegate::_OnSetFieldDictValueByKey': method with override specifier 'override' did not override any base class methods
5>d:\dev\usd\usd-qt\pxr\usdqt\undoStateDelegate.h(110): error C3668: 'pxrInternal_v0_19__pxrReserved__::UsdQt_UndoStateDelegate::_OnSetTimeSample': method with override specifier 'override' did not override any base class methods
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): error C2259: 'pxrInternal_v0_19__pxrReserved__::UsdQt_UndoStateDelegate': cannot instantiate abstract class
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: due to following members:
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetField(const pxrInternal_v0_19__pxrReserved__::SdfPath &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::SdfAbstractDataConstValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(187): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetField'
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetField(const pxrInternal_v0_19__pxrReserved__::SdfPath &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::VtValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(183): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetField'
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetFieldDictValueByKey(const pxrInternal_v0_19__pxrReserved__::SdfPath &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::SdfAbstractDataConstValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(198): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetFieldDictValueByKey'
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetFieldDictValueByKey(const pxrInternal_v0_19__pxrReserved__::SdfPath &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::TfToken &,const pxrInternal_v0_19__pxrReserved__::VtValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(193): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetFieldDictValueByKey'
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetTimeSample(const pxrInternal_v0_19__pxrReserved__::SdfPath &,double,const pxrInternal_v0_19__pxrReserved__::SdfAbstractDataConstValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(209): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetTimeSample'
5>D:\dev\usd\usd-qt\pxr\usdQt\undoStateDelegate.cpp(40): note: 'void pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetTimeSample(const pxrInternal_v0_19__pxrReserved__::SdfPath &,double,const pxrInternal_v0_19__pxrReserved__::VtValue &)': is abstract
5>D:\dev\usd\build\include\pxr/usd/sdf/layerStateDelegate.h(205): note: see declaration of 'pxrInternal_v0_19__pxrReserved__::SdfLayerStateDelegateBase::_OnSetTimeSample'
```

I get the same output when building through `cmake --build %BUILD_DIR% --config Release`

---

It's very likely I did something wrong regarding the cmake or build step. Any tips would be very welcome!

For completeness sake, I built `USD` using:
```bat
%PYTHON% build_scripts/build_usd.py --build-args boost,"--with-date_time --with-thread --with-system --with-filesystem" --no-maya %USD_BUILD_DIR%
```

--- a/pxr/usdQt/undoStateDelegate.cpp
+++ b/pxr/usdQt/undoStateDelegate.cpp
@@ -76,7 +76,7 @@ bool UsdQt_UndoStateDelegate::_InvertSetField(const SdfPath& path,
         TF_CODING_ERROR("Cannot invert field for expired layer.");
         return false;
     }
-    SetField(SdfAbstractDataSpecId(&path), fieldName, inverse);
+    SetField(path, fieldName, inverse);
     return true;
 }
 
@@ -91,7 +91,7 @@ bool UsdQt_UndoStateDelegate::_InvertSetFieldDictValueByKey(
             "Cannot invert field dictionary value for expired layer.");
         return false;
     }
-    SetFieldDictValueByKey(SdfAbstractDataSpecId(&path), fieldName, keyPath,
+    SetFieldDictValueByKey(path, fieldName, keyPath,
                            inverse);
     return true;
 }
@@ -104,7 +104,7 @@ bool UsdQt_UndoStateDelegate::_InvertSetTimeSample(
         TF_CODING_ERROR("Cannot invert time sample for expired layer.");
         return false;
     }
-    SetTimeSample(SdfAbstractDataSpecId(&path), time, inverse);
+    SetTimeSample(path, time, inverse);
     return true;
 }
 
@@ -123,11 +123,11 @@ bool UsdQt_UndoStateDelegate::_InvertCreateSpec(const SdfPath& path,
 /// XXX: This is copied straight from Sd.  Should this be refactored and
 /// packaged as a part of Sdf?
 static void _CopySpec(const SdfAbstractData& src, SdfAbstractData* dst,
-                      const SdfAbstractDataSpecId& specId) {
-    dst->CreateSpec(specId, src.GetSpecType(specId));
+                      const SdfPath& path) {
+    dst->CreateSpec(path, src.GetSpecType(path));
 
-    std::vector<TfToken> fields = src.List(specId);
-    TF_FOR_ALL(i, fields) { dst->Set(specId, *i, src.Get(specId, *i)); }
+    std::vector<TfToken> fields = src.List(path);
+    TF_FOR_ALL(i, fields) { dst->Set(path, *i, src.Get(path, *i)); }
 }
 
 bool UsdQt_UndoStateDelegate::_InvertDeleteSpec(
@@ -147,8 +147,8 @@ bool UsdQt_UndoStateDelegate::_InvertDeleteSpec(
         explicit _SpecCopier(SdfAbstractData* dst_) : dst(dst_) {}
 
         virtual bool VisitSpec(const SdfAbstractData& src,
-                               const SdfAbstractDataSpecId& specId) {
-            _CopySpec(src, dst, specId);
+                               const SdfPath& path) {
+            _CopySpec(src, dst, path);
             return true;
         }
 
@@ -228,85 +228,85 @@ bool UsdQt_UndoStateDelegate::_InvertPopPathChild(
 }
 
 void UsdQt_UndoStateDelegate::_OnSetFieldImpl(
-    const SdfAbstractDataSpecId& id, const TfToken& fieldName) {
+    const SdfPath& path, const TfToken& fieldName) {
     TF_DEBUG(USDQT_DEBUG_UNDOSTATEDELEGATE)
         .Msg("Setting Field '%s' for Spec '%s'\n", fieldName.GetText(),
-             id.GetFullSpecPath().GetText());
+             path.GetText());
     _MarkCurrentStateAsDirty();
-    const VtValue inverseValue = _layer->GetField(id, fieldName);
+    const VtValue inverseValue = _layer->GetField(path, fieldName);
     _RouteInverse(std::bind(&UsdQt_UndoStateDelegate::_InvertSetField,
-                            this, id.GetFullSpecPath(), fieldName,
+                            this, path, fieldName,
                             inverseValue));
 }
 
 void UsdQt_UndoStateDelegate::_OnSetFieldDictValueByKeyImpl(
-    const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+    const SdfPath& path, const TfToken& fieldName,
     const TfToken& keyPath) {
     TF_DEBUG(USDQT_DEBUG_UNDOSTATEDELEGATE)
         .Msg("Setting Dictionary Field '%s' By Key '%s' for Spec '%s'\n",
              fieldName.GetText(), keyPath.GetText(),
-             id.GetFullSpecPath().GetText());
+			 path.GetText());
     _MarkCurrentStateAsDirty();
     const VtValue inverseValue =
-        _layer->GetFieldDictValueByKey(id, fieldName, keyPath);
+        _layer->GetFieldDictValueByKey(path, fieldName, keyPath);
     _RouteInverse(std::bind(
         &UsdQt_UndoStateDelegate::_InvertSetFieldDictValueByKey, this,
-        id.GetFullSpecPath(), fieldName, keyPath, inverseValue));
+		path, fieldName, keyPath, inverseValue));
 }
 
 void UsdQt_UndoStateDelegate::_OnSetTimeSampleImpl(
-    const SdfAbstractDataSpecId& id, double time) {
+    const SdfPath& path, double time) {
     TF_DEBUG(USDQT_DEBUG_UNDOSTATEDELEGATE)
         .Msg("Setting Time Sample '%f' for Spec '%s'\n", time,
-             id.GetFullSpecPath().GetText());
+             path.GetText());
     _MarkCurrentStateAsDirty();
 
-    if (!_GetLayer()->HasField(id, SdfFieldKeys->TimeSamples)) {
+    if (!_GetLayer()->HasField(path, SdfFieldKeys->TimeSamples)) {
         _RouteInverse(std::bind(&UsdQt_UndoStateDelegate::_InvertSetField,
-                                this, id.GetFullSpecPath(),
+                                this, path,
                                 SdfFieldKeys->TimeSamples, VtValue()));
     } else {
         VtValue oldValue;
-        _GetLayer()->QueryTimeSample(id, time, &oldValue);
+        _GetLayer()->QueryTimeSample(path, time, &oldValue);
         _RouteInverse(
             std::bind(&UsdQt_UndoStateDelegate::_InvertSetTimeSample, this,
-                      id.GetFullSpecPath(), time, oldValue));
+                      path, time, oldValue));
     }
 }
 
-void UsdQt_UndoStateDelegate::_OnSetField(const SdfAbstractDataSpecId& id,
+void UsdQt_UndoStateDelegate::_OnSetField(const SdfPath& path,
                                                const TfToken& fieldName,
                                                const VtValue& value) {
-    _OnSetFieldImpl(id, fieldName);
+    _OnSetFieldImpl(path, fieldName);
 }
 
 void UsdQt_UndoStateDelegate::_OnSetField(
-    const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+    const SdfPath& path, const TfToken& fieldName,
     const SdfAbstractDataConstValue& value) {
-    _OnSetFieldImpl(id, fieldName);
+    _OnSetFieldImpl(path, fieldName);
 }
 
 void UsdQt_UndoStateDelegate::_OnSetFieldDictValueByKey(
-    const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+    const SdfPath& path, const TfToken& fieldName,
     const TfToken& keyPath, const VtValue& value) {
-    _OnSetFieldDictValueByKeyImpl(id, fieldName, keyPath);
+    _OnSetFieldDictValueByKeyImpl(path, fieldName, keyPath);
 }
 
 void UsdQt_UndoStateDelegate::_OnSetFieldDictValueByKey(
-    const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+    const SdfPath& path, const TfToken& fieldName,
     const TfToken& keyPath, const SdfAbstractDataConstValue& value) {
-    _OnSetFieldDictValueByKeyImpl(id, fieldName, keyPath);
+    _OnSetFieldDictValueByKeyImpl(path, fieldName, keyPath);
 }
 
 void UsdQt_UndoStateDelegate::_OnSetTimeSample(
-    const SdfAbstractDataSpecId& id, double time, const VtValue& value) {
-    _OnSetTimeSampleImpl(id, time);
+    const SdfPath& path, double time, const VtValue& value) {
+    _OnSetTimeSampleImpl(path, time);
 }
 
 void UsdQt_UndoStateDelegate::_OnSetTimeSample(
-    const SdfAbstractDataSpecId& id, double time,
+    const SdfPath& path, double time,
     const SdfAbstractDataConstValue& value) {
-    _OnSetTimeSampleImpl(id, time);
+    _OnSetTimeSampleImpl(path, time);
 }
 
 void UsdQt_UndoStateDelegate::_OnCreateSpec(const SdfPath& path,
@@ -323,7 +323,7 @@ void UsdQt_UndoStateDelegate::_OnCreateSpec(const SdfPath& path,
 // XXX: This is copied from SdLayer
 static void _CopySpecAtPath(const SdfAbstractData& src, SdfAbstractData* dst,
                             const SdfPath& path) {
-    _CopySpec(src, dst, SdfAbstractDataSpecId(&path));
+    _CopySpec(src, dst, path);
 }
 
 void UsdQt_UndoStateDelegate::_OnDeleteSpec(const SdfPath& path,

--- a/pxr/usdQt/undoStateDelegate.h
+++ b/pxr/usdQt/undoStateDelegate.h
@@ -87,33 +87,33 @@ private:
 
     void _OnSetLayer(const SdfLayerHandle& layer) override;
 
-    void _OnSetField(const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+    void _OnSetField(const SdfPath& path, const TfToken& fieldName,
                      const VtValue& value) override;
-    virtual void _OnSetField(const SdfAbstractDataSpecId& id,
+    virtual void _OnSetField(const SdfPath& path,
                              const TfToken& fieldName,
                              const SdfAbstractDataConstValue& value) override;
-    void _OnSetFieldImpl(const SdfAbstractDataSpecId& id,
+    void _OnSetFieldImpl(const SdfPath& path,
                          const TfToken& fieldName);
 
-    virtual void _OnSetFieldDictValueByKey(const SdfAbstractDataSpecId& id,
+    virtual void _OnSetFieldDictValueByKey(const SdfPath& path,
                                            const TfToken& fieldName,
                                            const TfToken& keyPath,
                                            const VtValue& value) override;
     virtual void _OnSetFieldDictValueByKey(
-        const SdfAbstractDataSpecId& id, const TfToken& fieldName,
+        const SdfPath& path, const TfToken& fieldName,
         const TfToken& keyPath,
         const SdfAbstractDataConstValue& value) override;
-    void _OnSetFieldDictValueByKeyImpl(const SdfAbstractDataSpecId& id,
+    void _OnSetFieldDictValueByKeyImpl(const SdfPath& path,
                                        const TfToken& fieldName,
                                        const TfToken& keyPath);
 
-    virtual void _OnSetTimeSample(const SdfAbstractDataSpecId& id, double time,
+    virtual void _OnSetTimeSample(const SdfPath& path, double time,
                                   const VtValue& value) override;
 
     virtual void _OnSetTimeSample(
-        const SdfAbstractDataSpecId& id, double time,
+        const SdfPath& path, double time,
         const SdfAbstractDataConstValue& value) override;
-    void _OnSetTimeSampleImpl(const SdfAbstractDataSpecId& id, double time);
+    void _OnSetTimeSampleImpl(const SdfPath& path, double time);
 
     virtual void _OnCreateSpec(const SdfPath& path, SdfSpecType specType,
                                bool inert) override;

